### PR TITLE
fix few warning builds.

### PR DESCRIPTION
**Pull request by devnexen**

### CODE REVIEW

File: src/backends/unix/system.c [hunk 1]
```diff
@@ -402,7 +402,8 @@ Sys_UnloadGame(void)
 void *
 Sys_GetGameAPI(void *parms)
 {
-	void *(*GetGameAPI)(void *);
+	typedef void *(*fnAPI)(void *);
+	fnAPI GetGameAPI;
```
Comment (Yamagi) [inline src/backends/unix/system.c:406]: What warning does this fix?

Comment (devnexen) [reply src/backends/unix/system.c:406]: This particular line does not however it s to fix the dlsym part below

```
warning: ISO C forbids assignment between function pointer and ‘void *’
482 |         GetGameAPI = (void *)dlsym(game_library, "GetGameAPI");

```

Comment (Yamagi) [reply src/backends/unix/system.c:406]: Ah, I see.

**Comment by Yamagi** [comment]
Thank you :)

**Pull request merged**